src/wifi.rs

```diff
@@ -1192,6 +1192,37 @@ impl<'d> WifiDriver<'d> {
         Ok(())
     }
 
+    /// Enables or disables promiscuous mode for the [`WifiDriver`].
+    ///
+    /// When promiscuous mode is enabled, the driver captures all Wifi frames
+    /// on the network, regardless of their destination MAC address. This is useful for
+    /// debugging or monitoring purposes.
+    pub fn set_promiscuous(&mut self, state: bool) -> Result<(), EspError> {
+        esp!(unsafe { esp_wifi_set_promiscuous(state) })?;
+
+        if state {
+            log::info!("Driver set in promiscuous mode");
+        } else {
+            log::info!("Driver set in non-promiscuous mode");
+        }
+
+        Ok(())
+    }
+
+    /// Gets the state of the promiscuous mode for the [`WifiDriver`].
+    pub fn get_promiscuous(&self) -> Result<bool, EspError> {
```

Comment:
Nit: `is_promiscuous`?

Reply:
Sure, i was keeping the name of the C function below but this makes more sense

Reply:
Thanks; also in the changelog :p

Reply:
shit